### PR TITLE
rework getProject response

### PR DESCRIPTION
**Pull request by Glunky**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/840

### CODE REVIEW

**Comment by Glunky** [comment]
rebase will be later

**Review by Cheephot — APPROVED** [review]

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 1]
```diff
@@ -35,49 +37,56 @@ public class GetProjectCommand : IGetProjectCommand
...
             }
             catch (Exception exc)
             {
-                errors.Add(errorMessage);
-
                 _logger.LogError(exc, "Exception on get department request.");
             }
+            finally
+            {
+                errors.Add($"Can not get department info for DepartmentId '{departmentId}'. Please try again later.");
```
Comment (ndoriomedov) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:60]: Why it is not in "catch" block? In finally we will always get error message, even if there wouldn't any errors, won't we?

Comment (Glunky) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:60]: there is only one case to return success result - departmentResponse.IsSuccess.
other cases are errors

Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:65]: you can remove finally here, and add errors after catch. You don't need finally here

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 2]
```diff
@@ -140,23 +110,28 @@ private List<ImageInfo> GetImages(List<Guid> imageIds, List<string> errors)
...
+            if (!projectUsers.Any())
             {
-                List<Guid> userIds = projectUsers.Select(x => x.UserId).Distinct().ToList();
+                return projectUsersInfo;
+            }
 
-                errorMessage = $"Can not get users info for UserIds {string.Join('\n', userIds)}. Please try again later.";
+            List<Guid> userIds = projectUsers.Select(x => x.UserId).ToList();
 
+            try
```
Comment (BelehovEgor) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:122]: move to new private method

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs (hunk 1 above)
Comment (Glunky) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:65]: why? here is 2 places where we can get an errors - it catch and if response is not success. To not duplicate errors.add in 2 places i added finaly.

**Review by ndoriomedov — APPROVED** [review]

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs (hunk 1 above)
Comment (Glunky) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs]: ++

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs (hunk 2 above)
Comment (Glunky) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:121]: ++

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 3]
```diff
@@ -40,60 +42,26 @@ public class GetProjectCommand : IGetProjectCommand
...
-
-                if (departmentResponse.Message.IsSuccess)
+                IOperationResult<IGetDepartmentResponse> departmentResponse = 
+                    _departmentRequestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>
+                    (
+                        IGetDepartmentRequest.CreateObj(null, departmentId)
+                    )
+                    .Result.Message;
+
+                if (departmentResponse.IsSuccess)
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:54]: I think you should add check what departmentResponse. Body != null

**Review by BelehovEgor — APPROVED** [review]

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 4]
```diff
@@ -40,60 +42,26 @@ public class GetProjectCommand : IGetProjectCommand
...
                 {
-                    department = _departmentInfoMapper.Map(departmentResponse.Message.Body);
+                    return _departmentInfoMapper.Map(departmentResponse.Body);
                 }
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:57]: Please, add loger if !departmentResponse.IsSuccess and not exception

Comment (Glunky) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:57]: ++

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**